processor/src/main/java/org/mapstruct/ap/processor/MethodRetrievalProcessor.java

```diff
@@ -80,19 +93,60 @@ public int getPriority() {
         return 1;
     }
 
+    private List<SourceMethod> retrievePrototypeMethods(TypeMirror typeMirror, MapperConfig mapperConfig) {
+        if ( typeMirror == null || typeMirror.getKind() == TypeKind.VOID ) {
+            return Collections.emptyList();
+        }
+
+        TypeElement typeElement = asTypeElement( typeMirror );
+        List<SourceMethod> methods = new ArrayList<SourceMethod>();
+        for ( ExecutableElement executable : getAllEnclosedExecutableElements( elementUtils, typeElement ) ) {
+
+            ExecutableType methodType = typeFactory.getMethodType( typeElement, executable );
+            List<Parameter> parameters = typeFactory.getParameters( methodType, executable );
+            boolean containsTargetTypeParameter = SourceMethod.containsTargetTypeParameter( parameters );
+
+            // prototype methods don't have prototypes themselves
```

Comment:
>  I would say that an interface is normal behavior is to hide the implementation. So only have a 'create' and 'update' for the root level element and hiding all the other branch and leaf mappings would actually make sense. 

Dunno, not sure whether I'd subscribe to that for different mapping methods for one model. I don't see an actual case where one really would need this sort of thing. Esp. if it comes to iterable mappings and their element mapping method.

Btw. you already can achieve that if really needed by declaring your "internal" mappings on another mapper which you don't expose and import via `uses()`.

> So, take for instance OSGi why couldn't only the Mapper be exported and we keep the generated implementation and the MapperConfig internal?

Ah, yes, that's something I'd definitely find usefulness :) We should actually provide a way for generating mapper implementations in a specific package to facilitate this case. Could you open an issue for that? Thx!